pyvrp/Population.py

```diff
@@ -2,18 +2,49 @@
 
 from bisect import insort_left
 from dataclasses import dataclass
-from typing import Callable, Iterator, List, NamedTuple, Tuple
+from typing import (
+    Callable,
+    Generic,
+    Iterator,
+    List,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from pyvrp.diversity.BiasedFitness import BiasedFitness
 
 from .Individual import Individual
 from .PenaltyManager import PenaltyManager
 from .ProblemData import ProblemData
 from .XorShift128 import XorShift128
 
-_DiversityMeasure = Callable[[ProblemData, Individual, Individual], float]
+
+class PopulationIndividual(Protocol):  # pragma: no cover
+    """
+    Defines an individual that can be put in a population target protocol.
+    """
+
+    def cost(self) -> Union[int, float]:
```

Comment:
`float` already covers `int`, so the union type is AFAIK unnecessary.

Reply:
I copied it from here so maybe then also remove it there: https://github.com/N-Wouda/PyVRP/blob/e04aba4a0849a7c9579eac1e76d1283bf3e67170/pyvrp/OptimisationTarget.py#L9

Reply:
Yes, thanks!